src/main/java/org/spongepowered/common/service/pagination/SpongePaginationService.java

```diff
@@ -0,0 +1,216 @@
+        };
+
+        final Command.Parameterized page = Command.builder()
+                .setShortDescription(TextComponent.of("Go to a specific page"))
+                .parameter(pageParameter)
+                .setExecutor(pageExecutor)
+                .build();
+
+        //We create the child manually in order to force that paginationElement is required for all children + fallback
+        //https://github.com/SpongePowered/SpongeAPI/issues/1272
+        return Command.builder()
+                .parameters(paginationIdParameter, Parameter.firstOf(pageParameter,
+                        Parameter.subcommand(next, "next", "n"),
+                        Parameter.subcommand(prev, "prev", "p", "previous"),
+                        Parameter.subcommand(page, "page")))
+                .child(page, "page")
+                .setExecutor(page)
+                .setShortDescription(TextComponent.of("Helper command for paginations occurring"))
+                .build();
+    }
+
+    private final class ActivePaginationParameter implements ValueParameter<ActivePagination> {
+
+        @Override
+        public Optional<? extends ActivePagination> getValue(final Parameter.Key<? super ActivePagination> parameterKey,
+                final ArgumentReader.Mutable reader,
+                final CommandContext.Builder context) throws ArgumentParseException {
+            final Audience source = context.getCause().first(Audience.class)
+                    .orElseThrow(() -> reader.createException(TextComponent.of("No usable source found")));
+
+            final SourcePaginations paginations = SpongePaginationService.this.getPaginationState(source, false);
+            if (paginations == null) {
+                final String name = source instanceof Nameable ? ((Nameable) source).getName() : source.toString();
+                throw reader.createException(TextComponent.of(String.format("Source %s has no paginations!", name)));
+            }
+
+            final UUID id;
+            final ArgumentReader.Immutable state = reader.getImmutable();
+            try {
+                id = UUID.fromString(reader.parseString());
+            } catch (final IllegalArgumentException ex) {
+                if (paginations.getLastUuid() != null) {
+                    reader.setState(state);
+                    return Optional.ofNullable(paginations.get(paginations.getLastUuid()));
+                }
+                throw reader.createException(TextComponent.of("Input was not a valid UUID!"));
+            }
+            final ActivePagination pagination = paginations.get(id);
+            if (pagination == null) {
+                throw reader.createException(TextComponent.of("No pagination registered for id " + id));
+            }
+            return Optional.ofNullable(paginations.get(id));
+        }
+
+        @Override
+        public List<String> complete(final CommandContext context) {
+            return context.getCause().first(Audience.class)
```

Comment:
`.first(Audience.class)` -> `.getAudience()` (audiences do have special handling in command causes)